src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -474,11 +475,16 @@ def _handle_grpc_error(
         on_error: Callable[[Exception], None],
     ):
         grpc_error_code = grpc_error.code()
-        error = FetchX509SvidError(str(grpc_error_code))
-        on_error(error)
 
         if retry_handler and grpc_error_code not in _NON_RETRYABLE_CODES:
+            logging.error(
+                'Error connecting to the Workload API: {}'.format(str(grpc_error_code))
+            )
```

Comment:
I don't know what options you are referring to.

Reply:
Here we are using the default logging module directly. I think we should create our own logger and configure it.
By default, the `logging` module logs the messages with a severity level of `WARNING` or above. This does not impact here because logs are for error and exception but if we want to add logs for debug purposes we need to set the _LogLevel_ accordingly. 
The default format for those messages is: `LEVEL:logger:message`. I think we can add more contexts to the log statements like time, module or line number. Then if necessary we can also write logs in json format. There are several options to consider.

Reply:
Can you create a ticket to address the logging properties ?

Reply:
Sure, this is the issue: https://github.com/HewlettPackard/py-spiffe/issues/95